access_error.go

```diff
@@ -34,10 +34,19 @@ func (f *Fosite) WriteAccessError(rw http.ResponseWriter, _ AccessRequester, err
 func (f *Fosite) writeJsonError(rw http.ResponseWriter, err error) {
 	rw.Header().Set("Content-Type", "application/json;charset=UTF-8")
 
-	rfcerr := ErrorToRFC6749Error(err)
+	// We dereference to make a copy and do not change passed err when it is already RFC6749Error.
+	rfcerr := *ErrorToRFC6749Error(err)
 	if !f.SendDebugMessagesToClients {
 		rfcerr.Debug = ""
 	}
+	// We expose both error hint and debug strings through standard error description, too
+	// (they are non-standard fields and some clients do not show them).
+	if rfcerr.Hint != "" {
+		rfcerr.Description += ": " + rfcerr.Hint
+	}
+	if rfcerr.Debug != "" {
+		rfcerr.Description += " (" + rfcerr.Debug + ")"
```

Comment:
So debug might contain internal information such as `UNIQUE CONSTRAINT VIOLATION` or other database or sensitive details. Let's keep the debug semantics as is :)

Reply:
Oh, I see - https://github.com/ory/fosite/pull/460/files#diff-2ac6e970b8a496524d4443819d71fa8cR62 solves this by checking the config. We should do that here as well!

Reply:
We are, above we set:

```
	if !f.SendDebugMessagesToClients {
		rfcerr.Debug = ""
	}
```

Reply:
Ah I missed that!